renderdoc/driver/vulkan/vk_core.cpp

```diff
@@ -4811,37 +4823,27 @@ bool WrappedVulkan::ShouldUpdateRenderState(ResourceId cmdid, bool forcePrimary)
   // if forcePrimary is set we're tracking renderpass activity that only happens in the primary
   // command buffer. So even if a secondary is partial, we still want to check it.
   if(forcePrimary)
-    return m_Partial[Primary].partialParent == cmdid;
+    return (cmdid == m_Partial.currentPartial) && m_Partial.isPartialPrimary;
 
   // otherwise, if a secondary command buffer is partial we want to *ignore* any state setting
   // happening in the primary buffer as fortunately no state is inherited (so we don't need to
   // worry about any state before the execute) and any state setting recorded afterwards would
   // incorrectly override what we have.
-  if(m_Partial[Secondary].partialParent != ResourceId())
-    return cmdid == m_Partial[Secondary].partialParent;
-
-  return cmdid == m_Partial[Primary].partialParent;
+  return cmdid == m_Partial.currentPartial;
```

Comment:
As an aside, I went digging on this function because it seemed very wrong only to realise that it doesn't really do what it says anymore, and these comments are inaccurate.

From history diving the relevant commits are 41d1acc47a040bd009d9d24f105a07d99413d4b2 and ef2804bfd30898816409f9a6541a1448969cafb5. Previously we used this function to do as it says - check if the renderstate should be updated (only for the innermost partial command buffer unless forced to the primary, as this does), but then was refactored in favour of per-commandbuffer renderstate tracking which could be done unconditionally.

I would suggest this function should be renamed to `IsPartialCommandBuffer` and perhaps even split out the `forcePrimary` to a separate function like `IsPrimaryPartialCommandBuffer`, since it's only now used for tracking renderpass activity, and not state, so the name is quite misleading.